### PR TITLE
Fix MSBuild CI jobs + minor cmake fixes

### PR DESCRIPTION
**Pull request by rkitover**
- Fix the MSBuild CI jobs #1278 
- Update blake3 from 1.3.1 to 71a2646
- Fix cmake dev warning when downloading zstd/hiredis
- Add a basic CMakeSettings.json for Visual Studio IDE with Ninja as the generator

### CODE REVIEW

File: cmake/Findhiredis.cmake [hunk 1]
```diff
@@ -60,6 +60,7 @@ if(do_download)
     URL https://github.com/redis/hiredis/archive/refs/tags/v${hiredis_version}.tar.gz
     SOURCE_DIR ${hiredis_dir}
     BINARY_DIR ${hiredis_build}
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
```
Comment (jrosdahl) [inline cmake/Findhiredis.cmake:63]: Looks like this was introduced in CMake 3.24? What happens if the user has an older CMake?

File: src/third_party/blake3/blake3_avx2_x86-64_windows_gnu.S [hunk 2]
```diff
@@ -1784,7 +1784,7 @@ blake3_hash_many_avx2:
         vmovdqu xmmword ptr [rbx+0x10], xmm1
         jmp     4b
 
-.section .rodata
+.section .rdata
```
[the remark this answers is not in the document]
Comment (rkitover) [reply src/third_party/blake3/blake3_avx2_x86-64_windows_gnu.S:1787]: Will fix.

File: cmake/Findhiredis.cmake (hunk 1 above)
Comment (rkitover) [reply cmake/Findhiredis.cmake:63]: It does in fact break cmake 3.18 on Debian 11, good catch there.

Comment (rkitover) [reply cmake/Findhiredis.cmake]: Fixed.

**Comment by jrosdahl** [comment]
Thanks.

**Pull request merged**